src/Smalot/PdfParser/PDFObject.php

```diff
@@ -274,14 +275,43 @@ public function getText(Page $page = null)
                         $args = preg_split('/\s/s', $command[self::COMMAND]);
                         $y = array_pop($args);
                         $x = array_pop($args);
+                        // get max char size and use as minimum for horizontal offset
+                        $fontMaxWidth = 0;
+                        if (null !== $current_font) {
+                            $fontDictonary = $current_font->getDetails();
+                            // type 0
+                            if (isset($fontDictonary['DescendantFonts'])) {
+                                $fontDictonary = $fontDictonary['DescendantFonts'][0];
+                            }
+                            // type 1
+                            if (isset($fontDictonary['Widths'])) {
+                                foreach ($fontDictonary['Widths'] as $width) {
+                                    if ((float) $width > $fontMaxWidth) {
+                                        $fontMaxWidth = (float) $width;
+                                    }
+                                }
+
+                                $fontMaxWidth = ($fontMaxWidth / 1000) * $current_font_size;
+                            }
+                            // CIDFontType2
+                            if ('cidfonttype2' === strtolower($fontDictonary['Type'])) {
+                                if (isset($fontDictonary['DW']) && $fontDictonary['DW']) {
+                                    $fontMaxWidth = ((float) $fontDictonary['DW'] / 1000) * $current_font_size;
+                                } else {
+                                    // default
+                                    $fontMaxWidth = $current_font_size;
+                                }
+                            }
+                        }
+
```

Comment:
* [ ] For **documentation** purposes: Can you add a comment via

```
/*
 * Get max char size and use as minimum for horizontal offset.
 */
```

at the beginning explaining the content of this if-clause in a few sentences. It would help to understand your code. For instance, why do you have a case for `CIDFontType2` and use `/ 1000` there?

* [ ] Please avoid using `(float) $width` again and again. Instead cast the variable, save the result and use that later on. This will avoid the trap of forgetting the cast sometimes.